ingest-app/src/cmr/ingest/validation/validation.clj

```diff
@@ -181,18 +182,45 @@
 (defn umm-spec-validate-collection
   "Validate collection through umm-spec validation functions. If warn? flag is
   true and umm-spec-validation is off, log warnings and return messages, otherwise throw errors."
-  [collection validation-options context warn?]
-  (when-let [err-messages (seq (umm-spec-validation/validate-collection
-                                collection
-                                (when (:validate-keywords? validation-options)
-                                  [(keyword-validations context)])))]
-    (if (or (:validate-umm? validation-options)
-            (config/return-umm-spec-validation-errors)
-            (not warn?))
-      (errors/throw-service-errors :invalid-data err-messages)
-      (do
-        (warn "UMM-C UMM Spec Validation Errors: " (pr-str (vec err-messages)))
-        err-messages))))
+  ([collection validation-options context warn?]
+   (umm-spec-validate-collection collection nil validation-options context warn?))
+  ([collection prev-collection validation-options context warn?]
+   (when-let [err-messages (seq (umm-spec-validation/validate-collection
+                                 collection
+                                 (when (:validate-keywords? validation-options)
+                                   [(keyword-validations context)])))]
+     (if (or (:validate-umm? validation-options)
+             (config/return-umm-spec-validation-errors)
+             (not warn?))
+       ;; whenever it's time to throw errors, we want to check if it's an collection update and
+       ;; it's not bulk-update and progressive-update-enabled is true. If so, we want to throw
+       ;; errors only when new errors are introduced, otherwise return all the existing errors as
+       ;; error-warnings.
+       (if (and (config/progressive-update-enabled)
+                (not (:bulk-update? validation-options)) 
+                prev-collection)
+         (let [prev-err-messages (if (and (:test-existing-errors? validation-options)
+                                          (= "mock-echo-system-token" (:token context)))
```

Comment:
This should be the configurable system token check, `cmr.transmit.config/echo-system-token?`.